sdk/textanalytics/Azure.AI.TextAnalytics/api/Azure.AI.TextAnalytics.netstandard2.0.cs

```diff
@@ -1736,9 +1732,9 @@ public static partial class TextAnalyticsModelFactory
         public static Azure.AI.TextAnalytics.ExtractKeyPhrasesResultCollection ExtractKeyPhrasesResultCollection(System.Collections.Generic.IEnumerable<Azure.AI.TextAnalytics.ExtractKeyPhrasesResult> list, Azure.AI.TextAnalytics.TextDocumentBatchStatistics statistics, string modelVersion) { throw null; }
         public static Azure.AI.TextAnalytics.ExtractSummaryActionResult ExtractSummaryActionResult(Azure.AI.TextAnalytics.ExtractSummaryResultCollection result, string actionName, System.DateTimeOffset completedOn) { throw null; }
         public static Azure.AI.TextAnalytics.ExtractSummaryActionResult ExtractSummaryActionResult(string actionName, System.DateTimeOffset completedOn, string code, string message) { throw null; }
-        public static Azure.AI.TextAnalytics.ExtractSummaryResult ExtractSummaryResult(string id, Azure.AI.TextAnalytics.TextDocumentStatistics statistics, Azure.AI.TextAnalytics.SummarySentenceCollection sentences) { throw null; }
+        public static Azure.AI.TextAnalytics.ExtractSummaryResult ExtractSummaryResult(string id, Azure.AI.TextAnalytics.TextDocumentStatistics statistics, System.Collections.Generic.IList<Azure.AI.TextAnalytics.SummarySentence> sentences, System.Collections.Generic.IList<Azure.AI.TextAnalytics.TextAnalyticsWarning> warnings = null) { throw null; }
```

Comment:
Should this use the IEnumerable type?

Reply:
In this case, this parameter is used to the `Sentences` property of the `ExtractSummaryResult` model. Because this is the property of a model, it uses `IList` instead of `IEnumerable` as per the guidelines here:
https://azure.github.io/azure-sdk/dotnet_introduction.html#dotnet-service-return-model-collections

Reply:
So in the output model, .NET uses either of IReadOnlyList or IList. But can use the more generic IEnumerable type for the input model?

Reply:
For both, input and output model properties, .NET prefers `IReadOnlyList` or `IList`. But for the method parameters in any other classes (e.g., client classes, helper classes, etc.), the use of `IEnumerable` is allowed if appropriate.